Automattic/jetpack
# AI Extension: introduce UI Handler context for the Jetpack Form AI Assistant feature

This PR introduces a react context to handle the AI extension in the UI. It aims to avoid propagating props down among nested components. The API is simple to use. The following is how it defines the `active` status of the toolbar button, depending on the assistant visibility state:

```jsx
export default function AiAssistantToolbarButton(): React.ReactElement {
	const { isVisible, toggle } = useContext( AiAssistantUiContext );

	return (
		<ToolbarButton
			showTooltip
			onClick={ toggle }
			aria-haspopup="true"
			aria-expanded={ true }
			label={ __( 'AI Assistant', 'jetpack' ) }
			icon={ aiAssistantIcon }
			disabled={ false }
			isActive={ isVisible }
		/>
	);
}
```

With this, we can start to handle the other pieces of the UI

Part of https://github.com/Automattic/jetpack/issues/32176
Fixes https://github.com/Automattic/jetpack/issues/32160


## Proposed changes:

* AI Extension: introduce UI Handler context for the Jetpack Form AI Assistant feature

### Other information:


## Jetpack product discussion



## Does this pull request change what data or activity we track or use?



## Testing instructions:






* Go to the block editor
* Ensure the `beta` extension is enabled
* Create a Jetpack Form block instance
* Click on the AI Button of the block toolbar
* Confirm it toggles its active status

https://github.com/Automattic/jetpack/assets/77539/e3c00994-d9e8-442f-bde0-37039d7c8554
